make_release.py

```diff
@@ -142,149 +163,185 @@ def short(rel_name):
+        return release_name[:pos] + '.0' + release_name[pos:]
+    return release_name + '.0'
+
+
+def pretag_pos(release_name):
+    """
+    gives the position of any pre-release tag
+    >>> pretag_pos('0.8')
+    >>> pretag_pos('0.8alpha25')
+    3
+    >>> pretag_pos('0.8.1rc1')
+    5
+    """
+    # 'a' needs to be searched for after 'beta'
+    for tag in ('rc', 'c', 'beta', 'b', 'alpha', 'a'):
+        match = re.search(tag + '\d+', release_name)
+        if match is not None:
+            return match.start()
+    return None
+
+
 def strip_pretags(release_name):
     """
     removes pre-release tags from a version string
-
-    >>> str(strip_pretags('0.8'))
+    >>> strip_pretags('0.8')
     '0.8'
-    >>> str(strip_pretags('0.8alpha25'))
+    >>> strip_pretags('0.8alpha25')
     '0.8'
-    >>> str(strip_pretags('0.8.1rc1'))
+    >>> strip_pretags('0.8.1rc1')
     '0.8.1'
     """
-    # 'a' needs to be searched for after 'beta'
-    for tag in ('rc', 'c', 'beta', 'b', 'alpha', 'a'):
-        release_name = re.sub(tag + '\d+', '', release_name)
-    return release_name
+    pos = pretag_pos(release_name)
+    return release_name[:pos] if pos is not None else release_name
 
 
 def isprerelease(release_name):
     """
     tests whether the release name contains any pre-release tag
-
     >>> isprerelease('0.8')
     False
     >>> isprerelease('0.8alpha25')
     True
     >>> isprerelease('0.8.1rc1')
     True
     """
-    return any(tag in release_name for tag in ('rc', 'c', 'beta', 'b', 'alpha', 'a'))
+    return pretag_pos(release_name) is not None
 
 
 # -------------------- #
 # end of generic tools #
 # -------------------- #
 
+# ------------------------- #
+# specific helper functions #
+# ------------------------- #
+
+
 def relname2fname(release_name):
     short_version = short(strip_pretags(release_name))
-    return r"version_%s.rst.inc" % short_version.replace('.', '_')
+    return r"version_{}.rst.inc".format(short_version.replace('.', '_'))
 
 
-def release_changes(release_name):
-    fpath = r"doc\source\changes\\" + relname2fname(release_name)
-    with open(fpath, encoding='utf-8-sig') as f:
+def release_changes(context):
+    directory = r"doc\source\changes"
+    fname = relname2fname(context['release_name'])
+    with open(os.path.join(context['build_dir'], directory, fname), encoding='utf-8-sig') as f:
         return f.read()
 
 
-def build_doc():
-    chdir('doc')
-    call('buildall.bat')
-    chdir('..')
-
-
-def update_changelog(release_name):
-    fname = relname2fname(release_name)
+def replace_lines(fpath, changes, end="\n"):
+    """
+    Parameters
+    ----------
+    changes : list of pairs
+        List of pairs (substring_to_find, new_line).
+    """
+    with open(fpath) as f:
+        lines = f.readlines()
+        for i, line in enumerate(lines[:]):
+            for substring_to_find, new_line in changes:
+                if substring_to_find in line and '#' not in line:
```

Comment:
I guess the `'#' not in line` is meant to avoid commented lines. In that case I would use something like `line.strip().startswith('#')` to avoid skipping a "used" line with a comment at the end of the line.